src/squeezeplay_squeezeos/share/applets/SetupTZ/SetupTZMeta.lua

```diff
@@ -27,23 +27,32 @@ function registerApplet(meta)
 	end
 end
 
-function notify_serverConnected(meta)
+function notify_serverConnected(meta, server)
 	-- On server connect, if the timezone still hasn't
-	-- been set, set it from SN's guess over http
+	-- been set, set it from LMS's guess over http
 	-- and unsubscribe if this succeeds
+
+	-- There is a small possibility of a race condition if more than one LMS
+	-- server becomes connected. But that won't cause harm.
 	if not squeezeos.getTimezone() then
-		local socket = SocketHttp(jnt, jnt:getSNHostname(), 80, "tzguess")
+		local socket = SocketHttp(jnt, server.ip, server.port, "tzguess")
```

Comment:
I've long lost touch with the firmware code... but wouldn't there be a helper method which wraps all these additional parameters?

Reply:
I didn't find any...